src/qsts3/settings.py

```diff
@@ -84,12 +84,25 @@
 # https://docs.djangoproject.com/en/5.0/ref/settings/#databases
 
 DATABASES = {
-    "default": {
-        "ENGINE": "django.db.backends.sqlite3",
-        "NAME": BASE_DIR / "db.sqlite3",
-    }
+   "default": {
```

Comment:
The idea of 12factor is to have your application code constant and only change things based on environment variables. Here you are introducing conditional logic just for the sake of customizing the set up.

It is better to simply add different environment variables (and use sensible defaults) here and keep your application code the same:

```
DATABASES = {
"DATABASE_ENGINE": os.getenv("QSTS_DATABASE_ENGINE", "django.db.backends.mysql"),
"DATABASE_HOST": os.getenv("QSTS_DATABASE_HOST", "mariadb"),
"DATABASE_PORT": os.getenv("QSTS_DATABASE_PORT", 3306),
"DATABASE_USER": os.getenv("QSTS_DATABASE_USER", "qsts"),
"DATABASE_PASSWORD": os.getenv("QSTS_DATABASE_PASSWORD"),
}
```